src/Stmt.h

```diff
@@ -525,7 +525,7 @@ class InitStmt final : public Stmt
 class NullStmt final : public Stmt
 	{
 public:
-	NullStmt() : Stmt(STMT_NULL) { }
+	NullStmt(StmtTag arg_tag = STMT_NULL) : Stmt(arg_tag) { }
```

Comment:
I'd rather not overwrite the tag. The tags are meant to identify the type of the statement, with "type" here corresponding to the class. By (optionally) setting this to something else, we're breaking the association (which could also cause trouble elsewhere, not sure).

 Two alternative suggestions:

1. Use a boolean or enum instead that indicates that this `NullStmt` reflects a directive, without changing its tag.
2. Create a new statement class `Directive` that then can return a corresponding tag (needs a check if other parts of the code now need to know about the new statement).